### ISSUE TITLE
libarchive dependency wrong?

### ISSUE THREAD

**Issue by bertsky** [tesseract-ocr/tesseract#3289]
If I build Tesseract with autotools, it drags in `libarchive` (`libarchive13` in my case), for the shared library as well as the CLI – in contrast to the prebuilt version.

But this fails for tesstrain-generated models:

```
archive_read_open_filename(...,/usr/local/share/tessdata/GT4HistOCR_2000000.traineddata,...) failed, Invalid or incomplete multibyte or wide character
```

So I tried to remove that dependency when calling `configure`, but none of the documented methods work, it always ends up giving me `checking for libarchive... yes`. So far, I have tried:
- `./configure --without-libarchive ...`
- `./configure --disable-libarchive ...`
- `libarchive_CFLAGS= libarchive_LIBS= ./configure ...`

What is the reason for the above failure, and is there any known workaround at build time?

**Comment by bertsky** [comment]
> But this fails for tesstrain-generated models:

Correction: it fails for all models.

**Comment by amitdo** [comment]
There no way to disable it currently.

Same for issue with libcurl.

**Comment by stweil** [comment]
Yes, there are currently no options to disable `libcurl` or `libarchive`.

The message listed above should only be shown for debug builds and is only an information. When opening a model file, Tesseract first tries whether it is a supported archive file (for example a zip or a compressed tar file). If that fails, it uses Tesseract's own proprietary format to access the components.

If the message is also shown for builds which were not configured with `--enable-debug`, that's an error.

In any case the message would disappear with model files using a standard archive format.

**Comment by stweil** [comment]
We could offer model files in one of the new formats. That would make the handling easier (for example adding or removing the dictionary) because it no longer requires `combine_tessdata`. The model files would also be much smaller.

But it would break compatibility with existing Tesseract binaries without libarchive support. And git cannot compress deltas for compressed files, so managing such model files with git would create large git repositories.

The new formats can safely be used by Linux distributions, for communities like OCR-D with fixed guidelines or for local use.

**Comment by bertsky** [comment]
Oh, I see. Thanks very much @stweil @amitdo for responding so quickly and explaining it!

Yes, it was a debug build in my case. Sorry, I should have checked myself whether the error actually does any harm.

IMHO backwards compatibility is extremely valuable in such cases...

**Comment by amitdo** [comment]
IMO, since we don't have an official tessdata repo with the new format, by default tesseract should not link with libarchive (even when it found on the system).

There should be a '--with-libarchive' option and 'with-libcurl'.

**Comment by amitdo** [comment]
https://autotools.io/autoconf/arguments.html

**Comment by amitdo** [comment]
.. and https://autotools.io/pkgconfig/pkg_check_modules.html

**Comment by Shreeshrii** [comment]
>There should be a '--with-libarchive' option and 'with-libcurl'.

I agree since libarchive is not required for the current set of traineddata files.

**Comment by stweil** [comment]
> by default tesseract should not link with libarchive.

We need a large user base with support for libarchive when we start providing model files using zip format. Therefore I disagree here. Ideally all distributions should link with libarchive and libcurl to support the maximum functionality. That's why our Windows and OCR-D use both libraries.

**Comment by stweil** [comment]
> There should be a '--with-libarchive' option and 'with-libcurl'.

Should it be `--with-archive` and `--with-curl`? The autoconf examples omit `lib`, and so do we for `libtensorflow` which already has `--with-tensorflow`.

**Comment by amitdo** [comment]
You can omit the 'lib' part.

I think Linux distro like Debianm Ubuntu and Fedora will enable any optional dependency anyway, unless it can be enable/disable them at run-time. Still for someone building from source downloaded directly from the repo we can disable it. Nevertheless, I accepts your preference.

**Comment by bertsky** [comment]
> > by default tesseract should not link with libarchive.
> 
> We need a large user base with support for libarchive when we start providing model files using zip format. Therefore I disagree here. Ideally all distributions should link with libarchive and libcurl to support the maximum functionality. That's why our Windows and OCR-D use both libraries.

So @stweil the move to std zip has already been decided? And the old format will still be supported by way of backwards compatibility? If so, perhaps the above warning should just be removed...

**Comment by stweil** [comment]
> If so, perhaps the above warning should just be removed...

I just tried, but failed to do so. See commit 3aae46d53dfd9a9623520a2b4b958bda9e361bc5 for the reason. :smile:

**Comment by Shreeshrii** [comment]
Homebrew CI on Github Actions gives an error even when libarchive has been installed.

```
checking for lept >= 1.74... yes
checking for libarchive... no
configure: error: in `/Users/runner/work/tesseract/tesseract':
configure: error: --with-archive was given, but test for libarchive failed
See `config.log' for more details
Error: Process completed with exit code 1.
```

See https://github.com/Shreeshrii/tesseract/runs/1878267170?check_suite_focus=true for complete log

**Comment by stweil** [comment]
`libarchive` is one of the Homebrew formulas which requires a path in `PKG_CONFIG_PATH`:

    For pkg-config to find libarchive you may need to set:
      export PKG_CONFIG_PATH="/opt/homebrew/opt/libarchive/lib/pkgconfig"

Use `export PKG_CONFIG_PATH="$(homebrew --config)/opt/libarchive/lib/pkgconfig"` (maybe with more directories added)  when running `configure`.

**Comment by Shreeshrii** [comment]
That job already had those defined:

```
    - name: Configure (MacOS Homebrew)
      run: |
           ./configure '--disable-shared' '--disable-openmp' '--disable-doc' '--with-pic' \
                'CC=${{ matrix.config.cc }}' \
                'CXX=${{ matrix.config.cxx }}' \
                'CXXFLAGS=-g -O2 -fsanitize=address,undefined' \
                'PKG_CONFIG_PATH=$(brew --prefix)/opt/icu4c/lib/pkgconfig:$(brew --prefix)/opt/libarchive/lib/pkgconfig:$(brew --prefix)/opt/libffi/lib/pkgconfig'
      if: runner.os == 'macOS'
```
Copied from the workflow linked from the runner log.
https://github.com/Shreeshrii/tesseract/actions/runs/557398627/workflow

**Comment by Shreeshrii** [comment]
It worked before the addition of new config options,

https://github.com/Shreeshrii/tesseract/runs/1877056112?check_suite_focus=true

**Comment by stweil** [comment]
I'll have a look later.

**Comment by stweil** [comment]
@Shreeshrii, replace `'...'` by `"..."` in the `configure` command. `$(brew --prefix)` won't work when it is enclosed with `'...'`.

**Comment by Shreeshrii** [comment]
@stweil Thank you! That fixed the issue.

**Comment by Shreeshrii** [comment]
The default values used in `Add new configure options --disable-doc, --with-archive and --with-curl #3292` mean that current builds where the dependencies are not installed will fail unless configure uses `--disable-doc --with-archive=no --with-curl=no`.
See last night's failing runs for [autotools](https://github.com/tesseract-ocr/tesseract/actions/runs/558972851) and [unittest](https://github.com/tesseract-ocr/tesseract/actions/runs/558816343) in https://github.com/tesseract-ocr/tesseract/actions

So we are adding four dependencies for the default build of tesseract :

- asciidoc
- xsltproc
- libarchive 
- libcurl 

If this is the intent, then documentation should be updated accordingly and another alpha release tagged to identify this change.

**Comment by stweil** [comment]
That's a bug. The default `check` should just do that: check what is available and only use that.

**Comment by Shreeshrii** [comment]
--with-archive and --with-curl are ok with the new commit. But the error regarding documentation is still there.

```
checking for asciidoc... false
checking for xsltproc... false
configure: error: in `/home/runner/work/tesseract/tesseract':
configure: error: --enable-doc was given, but test for asciidoc and xsltproc failed
See `config.log' for more details
```
Unittest run: https://github.com/tesseract-ocr/tesseract/runs/1889660797?check_suite_focus=true

It is ok in autotools run because I had used '--disable-doc' with configure.

```
Documentation will not be built because it was disabled.

```
Autotools run: https://github.com/tesseract-ocr/tesseract/runs/1890041190?check_suite_focus=true#step:6:158

**Comment by stweil** [comment]
Thanks for reporting. That's now fixed with commit b091b09b4f0a8cfa06166d249111190fb28712c4.

**Comment by zdenop** [comment]
Can issue be closed or are there any further problems?
